components/encoder/single/single_encoder_test.go

```diff
@@ -130,7 +129,7 @@ func TestEncoder(t *testing.T) {
 
 		ticks, _, err := enc.Position(context.Background(), encoder.PositionTypeUnspecified, nil)
 		test.That(t, err, test.ShouldBeNil)
-		test.That(t, ticks, test.ShouldEqual, 0)
```

Comment:
We should also test that the position has not changed if we have not received a tick, we can add a false Tick from the interrupt used in this test, and a secondary Position call that should stay at the same position.

Reply:
not sure if I understand what you mean here, a false tick (changing from high to low) is still a valid tick that will increase the position. We don't check if the tick is high or low in the encoder code.

Reply:
right, I forgot that we're double-counting for the single encoder because of historical reasons (we don't know what directions we're going, so get all changes of state in case). I don't agree with that reasoning, so I forget it all the time. Resolving.